src/App.tsx

```diff
@@ -1,73 +1,118 @@
-import React from 'react';
+import React, { useEffect, useState, useRef } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [inputValue, setInputValue] = useState('');
+  const [filteredPeople, setFilteredPeople] = useState(peopleFromServer);
+  const [selectedPerson, setSelectedPerson] = useState<null | {
+    name: string;
+    born: number;
+    died: number;
+  }>(null);
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const [debounceTimer, setDebounceTimer] = useState<NodeJS.Timeout | null>(
+    null,
+  );
+
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+
+  const debounceDelay = 300;
```

Comment:
The debounce delay is hardcoded as 300ms. According to the task requirements, this should be customizable via props. Consider adding a prop for debounce delay with a default value of 300ms.